params/astria_config.go

```diff
@@ -0,0 +1,369 @@
+
+		// Set fork-specific fields
+		orderedForks[i].Name = nh.name
+		orderedForks[i].Height = currentFork.Height
+		orderedForks[i].Halt = currentFork.Halt
+		orderedForks[i].SnapshotChecksum = ""
+
+		// Override with any new values from current fork
+		if currentFork.SnapshotChecksum != "" {
+			orderedForks[i].SnapshotChecksum = currentFork.SnapshotChecksum
+		}
+
+		if currentFork.ExtraDataOverride != nil {
+			orderedForks[i].ExtraDataOverride = currentFork.ExtraDataOverride
+		}
+
+		if currentFork.FeeCollector != nil {
+			orderedForks[i].FeeCollector = *currentFork.FeeCollector
+		}
+
+		if currentFork.EIP1559Params != nil {
+			orderedForks[i].EIP1559Params = *currentFork.EIP1559Params
+		}
+
+		if currentFork.Sequencer != nil {
+			orderedForks[i].Sequencer = *currentFork.Sequencer
+			orderedForks[i].Sequencer.RollupStartHeight = currentFork.Height
+			// set stop height for previous fork if sequencer data is changed
+			if i > 0 {
+				orderedForks[i-1].Sequencer.StopHeight = orderedForks[i-1].Sequencer.StartHeight + uint32(currentFork.Height-orderedForks[i-1].Sequencer.RollupStartHeight)
+			}
+		}
+
+		if currentFork.Celestia != nil {
+			orderedForks[i].Celestia = *currentFork.Celestia
+		}
+
+		if len(currentFork.BridgeAddresses) > 0 {
+			for _, cfg := range currentFork.BridgeAddresses {
+				err := cfg.Validate(orderedForks[i].Sequencer.AddressPrefix)
+				if err != nil {
+					return nil, fmt.Errorf("invalid bridge address config: %w", err)
+				}
+
+				if cfg.Erc20Asset == nil {
+					if nativeBridgeSeen {
+						return nil, errors.New("only one native bridge address is allowed")
+					}
+					nativeBridgeSeen = true
+				}
+
+				if cfg.Erc20Asset != nil && cfg.SenderAddress == (common.Address{}) {
+					return nil, errors.New("astria bridge sender address must be set for bridged ERC20 assets")
+				}
+
+				bridgeCfg := cfg
+				orderedForks[i].BridgeAddresses[cfg.BridgeAddress] = &bridgeCfg
+				orderedForks[i].BridgeAllowedAssets[cfg.AssetDenom] = struct{}{}
+				if cfg.Erc20Asset == nil {
+					log.Info("bridge for sequencer native asset initialized", "bridgeAddress", cfg.BridgeAddress, "assetDenom", cfg.AssetDenom)
+				} else {
+					log.Info("bridge for ERC20 asset initialized", "bridgeAddress", cfg.BridgeAddress, "assetDenom", cfg.AssetDenom, "contractAddress", cfg.Erc20Asset.ContractAddress)
+				}
+			}
+		}
+	}
+
+	if err := validateAstriaForks(orderedForks); err != nil {
+		return nil, err
+	}
+
+	// Sort orderedForks in descending order by Height
+	sort.Slice(orderedForks, func(i, j int) bool {
+		return orderedForks[i].Height > orderedForks[j].Height
+	})
+
+	return &AstriaForks{
+		orderedForks: orderedForks,
+		forkMap:      forks,
+	}, nil
+}
+
+func validateAstriaForks(forks []AstriaForkData) error {
+	for _, fork := range forks {
+		if !fork.Halt {
+			if fork.Sequencer.ChainID == "" {
+				return fmt.Errorf("fork %s: sequencer chain ID not set", fork.Name)
+			}
+
+			if fork.Sequencer.StartHeight == 0 {
```

Comment:
can multiple forks on the same sequencer network have the same start height? i'm guessing not

Reply:
yes they can because we also track the `RollupStartHeight`. Forks inherit the config from the previous fork and only need to define values that are changing. Sequencer `StartHeight` only needs to be re-defined and be different in a fork if the network is changing or if the rollup wants to skip some sequencer blocks for some reason. Most typical forks will not change the sequencer config and it will carry forward from fork to fork.